app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserAccountsAdapter.kt

```diff
@@ -101,30 +100,16 @@ class SwitchUserAccountsAdapter(
 
     override fun getItemCount(): Int = accounts.count()
 
-    fun upsertAccount(uiAccount: UiAccount, newCurrentMailboxObjectId: String?) {
-        val newList = createNewAccountsList(uiAccount)
-        isCollapsed = MutableList(newList.count()) { true }
-        isCollapsed[0] = false
-        currentMailboxObjectId = newCurrentMailboxObjectId
+    fun notifyAdapter(newList: List<UiAccount>) {
+        collapseAllExceptFirst(newList)
+        currentMailboxObjectId = MainViewModel.currentMailboxObjectId.value
         DiffUtil.calculateDiff(UiAccountsListDiffCallback(accounts, newList)).dispatchUpdatesTo(this)
         accounts = newList
     }
 
-    private fun createNewAccountsList(uiAccount: UiAccount): MutableList<UiAccount> {
-        val newList = accounts.toMutableList()
-        val existingAccount = newList.find { it.user.id == uiAccount.user.id }
-        if (existingAccount == null) {
-            newList.add(uiAccount)
-        } else {
-            existingAccount.mailboxes = uiAccount.mailboxes
-        }
-        return newList.sortAccounts()
-    }
-
-    private fun List<UiAccount>.sortAccounts(): MutableList<UiAccount> {
-        return filter { it.user.id != AccountUtils.currentUserId }
-            .toMutableList()
-            .apply { this@sortAccounts.find { it.user.id == AccountUtils.currentUserId }?.let { add(0, it) } }
+    private fun collapseAllExceptFirst(newList: List<UiAccount>) {
+        isCollapsed = MutableList(newList.count()) { true }
+        isCollapsed[0] = false
     }
```

Comment:
There is no need to do all the processing just to make a collapse, either use the `ViewHolder` or the `Model`.

Reply:
Fix in a new PR